actioncable/test/javascript/src/unit/connection_test.js

```diff
@@ -0,0 +1,28 @@
+import * as ActionCable from "../../../../app/javascript/action_cable/index"
+
+const {module, test} = QUnit
+
+module("ActionCable.Connection", () => {
+  module("#getState", () => {
+    test("uses the configured WebSocket adapter", assert => {
+      ActionCable.adapters.WebSocket = { foo: 1, BAR: "42" }
+      const connection = new ActionCable.Connection({})
+      connection.webSocket = {}
+      connection.webSocket.readyState = 1
+      assert.equal(connection.getState(), "foo")
+      connection.webSocket.readyState = "42"
+      assert.equal(connection.getState(), "bar")
+    })
+  })
+
+  module("#open", () => {
+    test("uses the configured WebSocket adapter", assert => {
+      const FakeWebSocket = function() {}
+      ActionCable.adapters.WebSocket = FakeWebSocket
+      const connection = new ActionCable.Connection({})
+      connection.monitor = { start() {} }
+      connection.open()
+      assert.equal(connection.webSocket instanceof FakeWebSocket, true)
+    })
+  })
+})
```

Comment:
Do these tests need to reset `ActionCable.adapters.WebSocket` to its original value?

Reply:
I believe that's handled by the `testDone` hook that you added in https://github.com/rails/rails/commit/410a32ffbdfa45aeee2c8ec0137201d3b95a33ff#diff-3f28927c13290f07f1f5db8933b6f74c:
https://github.com/rails/rails/blob/aa1ba9cb244b1e03d36aaa941ae4e91c6713b77e/actioncable/test/javascript/src/test_helpers/index.js#L9-L10
right?

Reply:
Oh right, glad I thought of that already. 🤦‍♂️